fondant/component_spec.py

```diff
@@ -71,11 +71,15 @@ def __repr__(self) -> str:
     def fields(self) -> t.Mapping[str, Field]:
         return types.MappingProxyType(
             {
-                name: Field(name=name, type=field["type"])
+                name: Field(name=name, type=Type[field["type"]])
```

Comment:
Was also wondering here, we return a dictionary with names and fields, but the fields themselves already contain a name. Is it worth having this twice?

Reply:
True. Returning a dict instead of a list allows lookup by name, which is definitely useful. Adding the name to the `Field`, makes it a complete representation of the field, which I believe is logical as well.

Currently we're not using the same approach for the `Subset`s though, so it would at least make sense to align on an approach.